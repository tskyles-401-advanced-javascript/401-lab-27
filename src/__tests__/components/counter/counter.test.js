import React from "react";
import Counter from '../../../components/counter/counter';

describe('<Counter/>', () => {
  it('is alive with correct markup at app start', () => {
    let app = shallow(<Counter/>);
    expect(app.find('.counter').exists()).toBeTruthy();
    expect(app.find('.down').exists()).toBeTruthy();
    expect(app.find('.up').exists()).toBeTruthy();
    expect(app.find('.count').exists()).toBeTruthy();
  });

  it('increase and decreases count when buttons are clicked', () => {
    let app = mount(<Counter/>);
    let buttonUp = app.find('.up');
    let buttonDown = app.find('.down');
    buttonUp.simulate('click');
    expect(app.state('count')).toBe(1);
    buttonDown.simulate('click');
    expect(app.state('count')).toBe(0);
  });

  it('changes polarity when count is positive or negative', () => {
    let app = mount(<Counter/>);
    let buttonUp = app.find('.up');
    let buttonDown = app.find('.down');
    buttonUp.simulate('click');
    expect(app.state('polarity')).toBe('positive');
    buttonDown.simulate('click');
    buttonDown.simulate('click');
    expect(app.state('polarity')).toBe('negative');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
});