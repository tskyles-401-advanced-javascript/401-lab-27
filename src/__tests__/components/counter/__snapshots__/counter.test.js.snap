// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
<section
  className="counter"
>
  <span
    className="down clicker"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count "
  >
    0
  </span>
  <span
    className="up clicker"
    onClick={[Function]}
  >
     + 
  </span>
</section>
`;
